speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/InterventionConfig.cpp

```diff
@@ -0,0 +1,55 @@
+#include <fstream>
+#include <string>
+#include <portability.h>
+
+#include "InterventionConfig.h"
+
+void InterventionConfig::SetInterventionAction(InterventionAction p_action) 
+{
+    m_currentAction = p_action;
+}
+
+void InterventionConfig::SetTextures(tTextureData* p_textures) 
+{
+    m_textures = p_textures;
+}
+
+tTextureData InterventionConfig::GetCurrentInterventionTexture() 
+{
+    return m_textures[m_currentAction];
```

Comment:
Should this make sure it doesn't index out of bound?

Reply:
fixed, checking for index being greater than array size (don't need to check negative numbers since it's an unsigned int)